src/banded/infreversecholeskytridiagonal.jl

```diff
@@ -0,0 +1,133 @@
+const MAX_TRIDIAG_CHOL_N = 2^21 - 1
+mutable struct LazySymTridiagonalReverseCholeskyFactor{T,M1,M2} <: BandedMatrices.AbstractBandedMatrix{T}
+    const A::M1 # original matrix
+    const L::M2 # LN
+    const ε::T  # adaptive tolerance
+    N::Int      # size of L used for approximating Ln     
+    n::Int      # size of approximated finite section
+end # this should behave like a lower Bidiagonal matrix
+function LazySymTridiagonalReverseCholeskyFactor(A, N, n, L, ε)
+    require_one_based_indexing(A)
+    M1, M2 = typeof(A), typeof(L)
+    T = eltype(L)
+    return LazySymTridiagonalReverseCholeskyFactor{T,M1,M2}(A, L, convert(T, ε), N, n)
+end
+
+function getproperty(C::ReverseCholesky{<:Any,<:LazySymTridiagonalReverseCholeskyFactor}, d::Symbol) # mimic getproperty(::ReverseCholesky{<:Any, <:Bidiagonal}, ::Symbol)
+    Cfactors = getfield(C, :factors)
+    #Cuplo    = 'L' 
+    if d == :U
+        return Cfactors'
+    elseif d == :L || d == :UL
+        return Cfactors
+    else
+        return getfield(C, d)
+    end
+end
+MemoryLayout(::Type{<:LazySymTridiagonalReverseCholeskyFactor}) = BidiagonalLayout{LazyLayout,LazyLayout}()
+
+size(L::LazySymTridiagonalReverseCholeskyFactor) = size(L.A)
+axes(L::LazySymTridiagonalReverseCholeskyFactor) = axes(L.A)
+Base.eltype(L::Type{LazySymTridiagonalReverseCholeskyFactor}) = eltype(L.L)
+
+copy(L::LazySymTridiagonalReverseCholeskyFactor) = LazySymTridiagonalReverseCholeskyFactor(copy(L.A), copy(L.L), L.ε, L.N, L.n)
+
+LazyBandedMatrices.bidiagonaluplo(L::LazySymTridiagonalReverseCholeskyFactor) = 'L'
+
+struct InfiniteBoundsAccessError <: Exception
+    i::Int
+    j::Int
+end
+function Base.showerror(io::IO, err::InfiniteBoundsAccessError)
+    print(io, "InfiniteBoundsAccessError: Tried to index reverse Cholesky factory at index (", err.i, ", ", err.j)
+    print(io, "), outside of the maximum allowable finite section of size (", MAX_TRIDIAG_CHOL_N, " × ", MAX_TRIDIAG_CHOL_N, ")")
+end
+
+function getindex(L::LazySymTridiagonalReverseCholeskyFactor, i::Int, j::Int)
+    max(i, j) > MAX_TRIDIAG_CHOL_N && throw(InfiniteBoundsAccessError(i, j))
+    T = eltype(L)
+    if j > i
+        return zero(T)
+    elseif max(i, j) > L.n
+        _expand_factor!(L, max(i, j))
+        return L.L[i, j]
+    else
+        return L.L[i, j]
+    end
+end
+
+function reversecholesky_layout(::SymTridiagonalLayout, ::NTuple{2,OneToInf{Int}}, A, ::NoPivot; tol=sqrt(1e-5), kwds...) # need a better tol...
```

Comment:
why is the tolerance so large?

Reply:
I had issues with the tolerance in a previous implementation but I realise now that was because I kept forcing it to resize no matter what... I've now changed it to `eps(real(T))` which seems to be fine.